### PR TITLE
Opfab cli : Improve file completion

### PR DESCRIPTION
**Pull request by freddidierRTE**
- In release notes :
  -  In chapter : Features 
  -  Text : #7277 Opfab CLI : Improve file completion

## Summary by CodeRabbit

- **New Features**
	- Introduced command-line completion functionality for the `opfab` command, enhancing usability across different shell environments (Zsh, Bash, etc.).
- **Refactor**
	- Simplified command structure for various operations, improving clarity and maintainability.
- **Chores**
	- Minor formatting adjustments in the `settings.gradle` file.
- **Improvements**
	- Streamlined installation process for `opfab-cli`, enhancing setup efficiency.

### CODE REVIEW

File: bin/load_environment_light.sh [hunk 1]
```diff
@@ -15,14 +15,11 @@
...
 nvm use v20.17.0
-
+echo "Install opfab-cli"
 (
-    echo "Install opfab-cli"
     cd $OF_HOME/cli/src
     npm install #Need first an install to download all necessary module
     npm install -g #Need global install to have the command opfab available
+    cd ..
+    source loadCompletion.sh
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply bin/load_environment_light.sh]: it is set in load_variables.sh

File: cli/loadCompletion.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  complete -F _opfab_completion opfab
+elif type compctl &>/dev/null; then
+  _opfab_completion () {
+    local cword line point si
+    read -Ac words
+    read -cn cword
+    read -l line
+    si="$IFS"
+    if ! IFS=$'
+' reply=($(opfab --compzsh --compgen "${cword}" "${words[cword-1]}" "${line}")); then
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply cli/loadCompletion.sh:35]: script auto generated via omelette , we avoid to modify it as much as possible

File: cli/loadCompletion.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Generated via opfab --completion
+# Added manually :  
+#    [[ $COMPREPLY ]] && return      
+#    _filedir
+# to solve issue with file completion see https://github.com/f/omelette/issues/4
+
+if type compdef &>/dev/null; then
+  _opfab_completion() {
+    compadd -- `opfab --compzsh --compgen "${CURRENT}" "${words[CURRENT-1]}" "${BUFFER}"`
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply cli/loadCompletion.sh:10]: script auto generated via omelette , we avoid to modify it as much as possible

File: cli/loadCompletion.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+elif type complete &>/dev/null; then
+  _opfab_completion() {
+    local cur prev nb_colon
+    _get_comp_words_by_ref -n : cur prev
+    nb_colon=$(grep -o ":" <<< "$COMP_LINE" | wc -l)
+
+    COMPREPLY=( $(compgen -W '$(opfab --compbash --compgen "$((COMP_CWORD - (nb_colon * 2)))" "$prev" "${COMP_LINE}")' -- "$cur") )
+    __ltrim_colon_completions "$cur"
+    # Added 
+    [[ $COMPREPLY ]] && return
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply cli/loadCompletion.sh]: Done

File: bin/load_environment_light.sh [hunk 5]
```diff
@@ -15,14 +15,11 @@ sdk install java 17.0.12-zulu
 sdk use java 17.0.12-zulu
 nvm install v20.17.0
 nvm use v20.17.0
-
+echo "Install opfab-cli"
 (
-    echo "Install opfab-cli"
     cd $OF_HOME/cli/src
     npm install #Need first an install to download all necessary module
     npm install -g #Need global install to have the command opfab available
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply bin/load_environment_light.sh:22]: Should never arise and any it is not a production script so it does not have to be perfect

**Review by ClementBouvierN — DISMISSED** [review]

**Review by ClementBouvierN — APPROVED** [review]

**Pull request merged**